### PR TITLE
[FEATURE]: SettingsAutocomplete component

### PR DESCRIPTION
**Pull request by juliepagano**
The purpose of this component is to normalize the rendering of settings within the panel options editors and other similar use cases.

This is a first draft of this component to make incremental progress. It is very likely the designs and some behaviors may change after syncing with designers

# Notes for reviewers

- I know we have some more work to do on the designs for this component. I have some time scheduled with design to work on that. However, I don't want to block incremental progress on this component (in service of getting table legend to have values) on those design adjustments, especially since this isn't making anything worse than it was before and in some cases is making an incremental improvement.
- Marking the component as "experimental" until the designs are more finalized.
- I'm writing pretty minimal tests here focused on the added logic because the vast majority of this component is MUI's `Autocomplete` and I don't want to exercise all the edge cases for the underlying library.
- I added some quick stories for the stat and gauge options editors to make it easier to test those changes.

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

# Screenshots

image
image

image
image
image

image

### CODE REVIEW

File: ui/panels-plugin/src/plugins/gauge-chart/GaugeChartOptionsEditorSettings.test.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ describe('GaugeChartOptionsEditorSettings', () => {
     const calcSelector = screen.getByRole('combobox', { name: 'Calculation' });
     userEvent.click(calcSelector);
     const meanOption = screen.getByRole('option', {
-      name: 'Mean',
+      name: /Mean/,
```
Comment (juliepagano) [inline ui/panels-plugin/src/plugins/gauge-chart/GaugeChartOptionsEditorSettings.test.tsx:69]: Switching these selectors to regex because the content of the `option` also includes descriptions now, and it would be annoying to test against the full thing, especially because descriptions may be more likely to be changed.

**Review by eunicorn — APPROVED** [review]
looks amazing!
